src/form/input/Input.tsx

```diff
@@ -49,32 +47,77 @@ export type InputProps = Omit<
   customClassName?: string;
   inputContainerRef?: React.RefObject<HTMLDivElement>;
   onChange: React.ReactEventHandler<HTMLInputElement>;
-  maxFractionDigits?: number;
+  localizationOptions?: {
+    shouldFormatToLocaleString?: boolean;
+    locale?: string;
+    maximumFractionDigits?: number;
+  };
 };
 
+/* eslint-disable complexity */
 function Input(props: InputProps) {
   const {
     testid,
+    value,
     type = "text",
     isDisabled,
     hasError,
     customClassName,
     leftIcon,
     rightIcon,
+    localizationOptions = {},
     role,
     autoComplete = "off",
     autoCorrect = "off",
     inputContainerRef,
     onChange,
-    maxFractionDigits = 0,
     ...rest
   } = props;
+  const {
+    shouldFormatToLocaleString = false,
+    locale,
+    maximumFractionDigits = 0
+  } = localizationOptions;
+  const [decimalSeparatorForLocale, setDecimalSeparatorForLocale] = useState(
+    () => getNumberSeparators(locale).DECIMAL_NUMBER_SEPARATOR
+  );
```

Comment:
As we are using both `DECIMAL_NUMBER_SEPARATOR` and `THOUSANDTHS_SEPARATOR`, let's store the whole thing within the state and then them below

```suggestion
  const [{
    DECIMAL_NUMBER_SEPARATOR: decimalSeparatorForLocale,
    THOUSANDTHS_SEPARATOR: thousandthSeparatorForLocale
  }, setNumberSeparatorsForLocale] = useState(
    () => getNumberSeparators(locale)
  );
```